pkgs/build-support/rust/build-rust-crate.nix

```diff
@@ -332,6 +332,9 @@ stdenv.mkDerivation rec {
     libName = if crate ? libName then crate.libName else crate.crateName;
     libPath = if crate ? libPath then crate.libPath else "";
 
+    # TODO: the documentation suggests this works, is this necessary?
+    preBuild = if lib.hasAttr "preBuild" crate then crate.preBuild else "";
```

Comment:
We need to rethink, how `crateOverrides` are applied here.

Reply:
Ideally it should work the same way as `defaultGemConfig`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/ruby-modules/bundled-common/default.nix#L38

Reply:
we would need to pass all arguments to the function defined by `cratesOverrides` before passing it to the underlying `stdenv.mkDerivation`

Reply:
If I remember correctly, my intention was to allow the user to override the derivation with standard Nix functions, but having multiple ways of doing this is not too bad. Wouldn't it look better with an inherit of multiple attributes instead of this extra line?

Reply:
I want this to configure the build of one of cargo-vendor's dependencies: libgit2-sys. I don't see how to use the standard Nix functions in this case. If I use `overrideAttrs`, I can produce a derivation for the libgit2-sys that will build but it wouldn't be used as a dependency, since the dependencies are assembled by carnix as part of a `rec` attribute map. If this were configurable like the overall package set, then it would be possible.

Even if it were possible to change the dependency using the standard methods, it seems like this is what `crateOverrides` are for. Any package using libgit2-sys requires this configuration, so crateOverrides should be able to specify more build attributes. I can do the fairly simple thing of `mkDerivation ({ ... } // crate)`, (removing things already processed) or I can whitelist particular attributes, or even make an extra attribute in crateOverrides for this purpose. Which one would you prefer?

Reply:
`mkDerivation ({ ... } // crate)` sounds like what I would expect from `crateOverrides`.